File: app/src/main/java/com/example/assignment/models/Chapter.java
package com.example.assignment.models;

import java.io.Serializable;

public class Chapter implements Serializable {
    private String id, title, content;
    private int index;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public Chapter() {
    }

    public Chapter(String id, String title, String content, int index) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.index = index;
    }
}

File: app/src/main/java/com/example/assignment/models/Chapter.kt
package com.example.assignment.models

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName
import java.io.Serializable

@Entity(
    tableName = "chapters",
    foreignKeys = [ForeignKey(
        entity = Comic::class,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("id_comic"),
        onDelete = ForeignKey.CASCADE
    )]
)
data class Chapter(
    @PrimaryKey var id: String,

    @SerializedName("title")
    var title: String,

    @SerializedName("content")
    var content: String,

    @SerializedName("chapter_index")
    var index: Int,

    @SerializedName("id_comic")
    var id_comic: String,

    @SerializedName("has_html")
    val hasHtml: Int
) : Serializable {

}